lib/Dialect/HW/Transforms/FlattenIO.cpp

```diff
@@ -187,7 +186,7 @@ class FlattenIOTypeConverter : public TypeConverter {
                                 ValueRange inputs, Location loc) {
       auto structType = getStructType(type);
       assert(structType && "expected struct type");
```

Comment:
@maerhart is right, we should just be able to drop this now, since we don't need the struct type anymore:
```suggestion
```